pkg/virt-handler/virtwrap/manager.go

```diff
@@ -416,7 +536,50 @@ func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
 	return &newSpec, nil
 }
 
+func (l *LibvirtDomainManager) RemoveVMSecrets(vm *v1.VM) error {
+	domName := VMNamespaceKeyFunc(vm)
+
+	// TODO this is horribly inefficient. Maybe there's a better way?
```

Comment:
@rmohr made a good observation offline about this.  Since virt-handler is the only consumer of the libvirt api, we can cache the secret list in a map to make this more efficient. 

I'll update the patch shortly to do this.